katpoint/ephem_extra.py

```diff
@@ -67,24 +67,39 @@ def _just_gimme_an_ascii_string(s):
         return str(s)
 
 
-def angle_from_degrees(s):
-    """Creates angle object from sexagesimal string in degrees or number in radians."""
+def _to_angle(s, unit='d'):
+    """Creates angle object from sexagesimal string with given `unit` or number in radians."""
+    # Try to obtain the unit from a string suffix (this is the v1 angle format)
+    try:
+        suffix = _just_gimme_an_ascii_string(s[-1])
+    except (TypeError, IndexError):
+        suffix = ''
+    if suffix in ('d', 'h'):
+        unit = suffix
+        s = s[:-1]
+    # Pick appropriate converter based on unit
+    if unit in ('d', 'deg', 'degrees'):
+        converter = ephem.degrees
+    elif unit in ('h', 'hour', 'hourangle', 'hours'):
```

Comment:
Is it necessary to accept multiple forms? A unit derived from the suffix can only be one character and this is a private function only used for angle_from_degrees and angle_from_hours so you don't have to cater to user's whims.

Reply:
True, will do.